.github/generate-native-matrix.sh

```diff
@@ -0,0 +1,29 @@
+#! /bin/bash
+
+IFS=$'\n'
+modules=($(find . -name pom.xml | sed -E 's@/pom.xml@@g' | sed -E 's@^\./@@g' | grep -vE '^\.$' | sort))
+count=${#modules[@]}
+partition=$(expr $count / 4 + 1)
+
+json=$'{\n'
+json+=$'    "include": [\n'
+json+=$'    {\n'
+i=0
+partition_index=1
+for module in ${modules[@]}; do
+    if [ $i -gt $partition ]; then
+        json+=$'"\n    },\n    {\n'
+        i=0
+    fi
+    if [ $i -eq 0 ]; then
+        json+="        \"category\": \"Native Tests - $partition_index\","
+        json+=$'\n        "test-modules": "'
+        ((partition_index=partition_index+1))
```

Comment:
`partition_index+=1` should work as well but doesn't really matter

Reply:
Tried it. Didn't work.

Reply:
Strange. So be it!